assets/log-server.ts

```diff
@@ -0,0 +1,80 @@
+import chalk, { Chalk } from 'chalk';
+import * as util from 'util';
+import * as ζws from 'ws';
+
+export interface ConsoleLogServerMessage {
+  category: 'console';
+  type: string;
+  data: any[];
+}
+
+export interface ConsoleLogServerOptions {
+  consolelogs: boolean;
+  consolelogsPort: number;
+}
+
+export function isConsoleLogServerMessage(m: any): m is ConsoleLogServerMessage {
+  return m
+    && typeof m.category === 'string'
+    && typeof m.type === 'string'
+    && m.data && typeof m.data.length === 'number';
+}
+
+export async function createConsoleLogServer(host: string, port: number): Promise<ζws.Server> {
+  const WebSocket = await import('ws');
+
+  const wss = new WebSocket.Server({ host, port });
+
+  wss.on('connection', ws => {
+    ws.on('message', data => {
+      let msg;
+
+      try {
+        data = data.toString();
+        msg = JSON.parse(data);
+      } catch (e) {
+        process.stderr.write(`Error parsing JSON message from client: "${data}" ${chalk.red(e.stack ? e.stack : e)}\n`);
+        return;
+      }
+
+      if (!isConsoleLogServerMessage(msg)) {
+        const m = util.inspect(msg, { colors: chalk.enabled });
+        process.stderr.write(`Bad format in client message: ${m}\n`);
+        return;
+      }
+
+      if (msg.category === 'console') {
+        let status: Chalk | undefined; // unknown levels are normal color
```

Comment:
Sorry to be picky--can we use the devkit colors? https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/core/src/terminal/colors.ts

Should look like this:

```ts
import { terminal } from '@angular-devkit/core';

status = terminal.red
```